Installing

I'm sure I'm trying to install something incorrectly. I'm not well versed in Go yet. Trying to download and install your project though and running into issues.

```bash
$ go get ./...
# github.com/astroband/astrologer/es
../../golang/src/github.com/astroband/astrologer/es/operation.go:144:7: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
../../golang/src/github.com/astroband/astrologer/es/operation.go:145:24: o.Body.MustManageOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustManageOfferOp)
../../golang/src/github.com/astroband/astrologer/es/operation.go:146:7: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
../../golang/src/github.com/astroband/astrologer/es/operation.go:147:31: o.Body.MustCreatePassiveOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustCreatePassiveOfferOp)
../../golang/src/github.com/astroband/astrologer/es/operation.go:191:23: undefined: "github.com/stellar/go/xdr".ManageOfferOp
../../golang/src/github.com/astroband/astrologer/es/operation.go:200:30: undefined: "github.com/stellar/go/xdr".CreatePassiveOfferOp
../../golang/src/github.com/astroband/astrologer/es/operation_result.go:22:8: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
../../golang/src/github.com/astroband/astrologer/es/operation_result.go:23:29: r.Tr.MustManageOfferResult undefined (type *"github.com/stellar/go/xdr".OperationResultTr has no field or method MustManageOfferResult)
../../golang/src/github.com/astroband/astrologer/es/operation_result.go:24:8: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
../../golang/src/github.com/astroband/astrologer/es/operation_result.go:74:29: undefined: "github.com/stellar/go/xdr".ManageOfferResult
../../golang/src/github.com/astroband/astrologer/es/operation_result.go:24:8: too many errors
# _/Users/Tyler/Desktop/astrologer/es
es/operation.go:144:7: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
es/operation.go:145:24: o.Body.MustManageOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustManageOfferOp)
es/operation.go:146:7: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
es/operation.go:147:31: o.Body.MustCreatePassiveOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustCreatePassiveOfferOp)
es/operation.go:191:23: undefined: "github.com/stellar/go/xdr".ManageOfferOp
es/operation.go:200:30: undefined: "github.com/stellar/go/xdr".CreatePassiveOfferOp
es/operation_result.go:22:8: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
es/operation_result.go:23:29: r.Tr.MustManageOfferResult undefined (type *"github.com/stellar/go/xdr".OperationResultTr has no field or method MustManageOfferResult)
es/operation_result.go:24:8: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
es/operation_result.go:74:29: undefined: "github.com/stellar/go/xdr".ManageOfferResult
es/operation_result.go:24:8: too many errors
```

So even under your branch I'm still getting errors trying to get the dependencies.

```bash
$ go get ./...
# github.com/astroband/astrologer/es
../../go/src/github.com/astroband/astrologer/es/operation.go:144:7: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
../../go/src/github.com/astroband/astrologer/es/operation.go:145:24: o.Body.MustManageOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustManageOfferOp)
../../go/src/github.com/astroband/astrologer/es/operation.go:146:7: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
../../go/src/github.com/astroband/astrologer/es/operation.go:147:31: o.Body.MustCreatePassiveOfferOp undefined (type "github.com/stellar/go/xdr".OperationBody has no field or method MustCreatePassiveOfferOp)
../../go/src/github.com/astroband/astrologer/es/operation.go:191:23: undefined: "github.com/stellar/go/xdr".ManageOfferOp
../../go/src/github.com/astroband/astrologer/es/operation.go:200:30: undefined: "github.com/stellar/go/xdr".CreatePassiveOfferOp
../../go/src/github.com/astroband/astrologer/es/operation_result.go:22:8: undefined: "github.com/stellar/go/xdr".OperationTypeManageOffer
../../go/src/github.com/astroband/astrologer/es/operation_result.go:23:29: r.Tr.MustManageOfferResult undefined (type *"github.com/stellar/go/xdr".OperationResultTr has no field or method MustManageOfferResult)
../../go/src/github.com/astroband/astrologer/es/operation_result.go:24:8: undefined: "github.com/stellar/go/xdr".OperationTypeCreatePassiveOffer
../../go/src/github.com/astroband/astrologer/es/operation_result.go:74:29: undefined: "github.com/stellar/go/xdr".ManageOfferResult
../../go/src/github.com/astroband/astrologer/es/operation_result.go:24:8: too many errors
```

--- a/es/operation.go
+++ b/es/operation.go
@@ -141,10 +141,12 @@ func NewOperation(t *Transaction, o *xdr.Operation, n byte) *Operation {
 		newPayment(o.Body.MustPaymentOp(), op)
 	case xdr.OperationTypePathPayment:
 		newPathPayment(o.Body.MustPathPaymentOp(), op)
-	case xdr.OperationTypeManageOffer:
-		newManageOffer(o.Body.MustManageOfferOp(), op)
-	case xdr.OperationTypeCreatePassiveOffer:
-		newCreatePassiveOffer(o.Body.MustCreatePassiveOfferOp(), op)
+	case xdr.OperationTypeManageSellOffer:
+		newManageSellOffer(o.Body.MustManageSellOfferOp(), op)
+	case xdr.OperationTypeManageBuyOffer:
+		newManageBuyOffer(o.Body.MustManageBuyOfferOp(), op)
+	case xdr.OperationTypeCreatePassiveSellOffer:
+		newCreatePassiveSellOffer(o.Body.MustCreatePassiveSellOfferOp(), op)
 	case xdr.OperationTypeSetOptions:
 		newSetOptions(o.Body.MustSetOptionsOp(), op)
 	case xdr.OperationTypeChangeTrust:
@@ -188,7 +190,7 @@ func newPathPayment(o xdr.PathPaymentOp, op *Operation) {
 	}
 }
 
-func newManageOffer(o xdr.ManageOfferOp, op *Operation) {
+func newManageSellOffer(o xdr.ManageSellOfferOp, op *Operation) {
 	op.SourceAmount = amount.String(o.Amount)
 	op.SourceAsset = NewAsset(&o.Buying)
 	op.OfferID = int(o.OfferId)
@@ -197,7 +199,16 @@ func newManageOffer(o xdr.ManageOfferOp, op *Operation) {
 	op.DestinationAsset = NewAsset(&o.Selling)
 }
 
-func newCreatePassiveOffer(o xdr.CreatePassiveOfferOp, op *Operation) {
+func newManageBuyOffer(o xdr.ManageBuyOfferOp, op *Operation) {
+	op.SourceAmount = amount.String(o.BuyAmount)
+	op.SourceAsset = NewAsset(&o.Selling)
+	op.DestinationAsset = NewAsset(&o.Buying)
+	op.OfferID = int(o.OfferId)
+	op.OfferPrice, _ = big.NewRat(int64(o.Price.N), int64(o.Price.D)).Float64()
+	op.OfferPriceND = &Price{int(o.Price.N), int(o.Price.D)}
+}
+
+func newCreatePassiveSellOffer(o xdr.CreatePassiveSellOfferOp, op *Operation) {
 	op.SourceAmount = amount.String(o.Amount)
 	op.SourceAsset = NewAsset(&o.Buying)
 	op.OfferPrice, _ = big.NewRat(int64(o.Price.N), int64(o.Price.D)).Float64()

--- a/es/operation_result.go
+++ b/es/operation_result.go
@@ -19,10 +19,12 @@ func AppendResult(op *Operation, r *xdr.OperationResult) {
 			newPaymentResult(r.Tr.MustPaymentResult(), op)
 		case xdr.OperationTypePathPayment:
 			newPathPaymentResult(r.Tr.MustPathPaymentResult(), op)
-		case xdr.OperationTypeManageOffer:
-			newManageOfferResult(r.Tr.MustManageOfferResult(), op)
-		case xdr.OperationTypeCreatePassiveOffer:
-			newManageOfferResult(r.Tr.MustCreatePassiveOfferResult(), op)
+		case xdr.OperationTypeManageSellOffer:
+			newManageSellOfferResult(r.Tr.MustManageSellOfferResult(), op)
+		case xdr.OperationTypeManageBuyOffer:
+			newManageBuyOfferResult(r.Tr.MustManageBuyOfferResult(), op)
+		case xdr.OperationTypeCreatePassiveSellOffer:
+			newManageSellOfferResult(r.Tr.MustCreatePassiveSellOfferResult(), op)
 		case xdr.OperationTypeSetOptions:
 			newSetOptionsResult(r.Tr.MustSetOptionsResult(), op)
 		case xdr.OperationTypeChangeTrust:
@@ -73,28 +75,41 @@ func newPathPaymentResult(r xdr.PathPaymentResult, op *Operation) {
 	}
 }
 
-func newManageOfferResult(r xdr.ManageOfferResult, op *Operation) {
+func newManageSellOfferResult(r xdr.ManageSellOfferResult, op *Operation) {
 	op.InnerResultCode = int(r.Code)
-	op.Succesful = r.Code == xdr.ManageOfferResultCodeManageOfferSuccess
+	op.Succesful = r.Code == xdr.ManageSellOfferResultCodeManageSellOfferSuccess
 
 	if s, ok := r.GetSuccess(); ok {
-		op.ResultOffersClaimed = appendOffersClaimed(s.OffersClaimed)
+		newManageOfferResult(s, op)
+	}
+}
 
-		if o, ok := s.Offer.GetOffer(); ok {
-			p, _ := big.NewRat(int64(o.Price.N), int64(o.Price.D)).Float64()
+func newManageBuyOfferResult(r xdr.ManageBuyOfferResult, op *Operation) {
+	op.InnerResultCode = int(r.Code)
+	op.Succesful = r.Code == xdr.ManageBuyOfferResultCodeManageBuyOfferSuccess
 
-			op.ResultOffer = &Offer{
-				Amount:   amount.String(o.Amount),
-				Price:    p,
-				PriceND:  Price{int(o.Price.N), int(o.Price.D)},
-				Buying:   *NewAsset(&o.Buying),
-				Selling:  *NewAsset(&o.Selling),
-				OfferID:  int64(o.OfferId),
-				SellerID: o.SellerId.Address(),
-			}
+	if s, ok := r.GetSuccess(); ok {
+		newManageOfferResult(s, op)
+	}
+}
 
-			op.ResultOfferEffect = s.Offer.Effect.String()
+func newManageOfferResult(s xdr.ManageOfferSuccessResult, op *Operation) {
+	op.ResultOffersClaimed = appendOffersClaimed(s.OffersClaimed)
+
+	if o, ok := s.Offer.GetOffer(); ok {
+		p, _ := big.NewRat(int64(o.Price.N), int64(o.Price.D)).Float64()
+
+		op.ResultOffer = &Offer{
+			Amount:   amount.String(o.Amount),
+			Price:    p,
+			PriceND:  Price{int(o.Price.N), int(o.Price.D)},
+			Buying:   *NewAsset(&o.Buying),
+			Selling:  *NewAsset(&o.Selling),
+			OfferID:  int64(o.OfferId),
+			SellerID: o.SellerId.Address(),
 		}
+
+		op.ResultOfferEffect = s.Offer.Effect.String()
 	}
 }
 

